pkgs/tools/networking/trurl/default.nix

```diff
@@ -1,27 +1,40 @@
-{ lib, stdenv, fetchFromGitHub, curl, perl }:
+{ lib, stdenv, fetchFromGitHub, installShellFiles, curl, python3 }:
 
 stdenv.mkDerivation rec {
   pname = "trurl";
-  version = "0.4";
+  version = "0.5";
 
   src = fetchFromGitHub {
     owner = "curl";
     repo = pname;
     rev = "${pname}-${version}";
-    sha256 = "sha256-f9z0gPtHcn3iWFA6MT6ngok0nXBcd6zJ8Tjnb5Lzf6c=";
+    sha256 = "sha256-ho/XEuLQoUPTQNUZpen7WdJf0Iz0RXHrh+VbbrS8c9c=";
   };
 
   separateDebugInfo = stdenv.isLinux;
 
   enableParallelBuilding = true;
 
+  preBuild = ''
+    # Fix up version number
+    substituteInPlace version.h --replace 0.4 0.5
+  '';
+
   buildInputs = [ curl ];
   makeFlags = [ "PREFIX=$(out)" ];
 
   doCheck = true;
-  checkInputs = [ perl ];
+  checkInputs = [ python3 ];
   checkTarget = "test";
 
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    # Remove wrong, hardcoded version in manpage
+    # https://github.com/curl/trurl/commit/49601415f9fd59b146893c98e47cdd89ebfd070f
+    substituteInPlace trurl.1 --replace "trurl 0.4" trurl
```

Comment:
please move this to `postPatch`

Reply:
Since I noticed the `installManPage` bit is not needed anyway (as that is already done by the `make install` step), I've decided to move this to prePatch as well, like the `version.h` fixup.

Hopefully that is okay, but happy to change it if the convention says otherwise.